src/treemodel/document.js

```diff
@@ -68,12 +74,26 @@ export default class Document {
 		this._pendingChanges = [];
 
 		/**
-		 * Selection done on this document.
+		 * List of roots that are owned and managed by this document. Use {@link core.treeModel.document#createRoot} and
+		 * {@link core.treeModel.document#getRoot} to manipulate it.
 		 *
 		 * @readonly
-		 * @member {core.treeModel.Selection} core.treeModel.Document#selection
+		 * @protected
+		 * @member {Map} core.treeModel.Document#roots
 		 */
-		this.selection = new Selection();
+		this._roots = new Map();
+
+		// Add events that will update selection attributes.
+		this.selection.on( 'update', () => {
```

Comment:
I'm not sure on what events the selection should update its attributes itself. This should not conflict with the situation when the feature sets the selection and its attributes (e.g. bold command when executed will add attributes to the selection and that will, if I understand the flow correctly, trigger `#changesDone`).